### PR TITLE
SNOW-847264: Use centos7 with node14 image

### PR DESCRIPTION
**Pull request by sfc-gh-dprzybysz**
### Description: 
Use centos7 with node14 image

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

**Comment by sfc-gh-dprzybysz** [comment]
Jenkins jobs definition should be merged first to make jenkins build successful https://github.com/snowflakedb/jenkins_utils/pull/8053

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -109,16 +109,10 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                image: [ 'nodejs-centos6-default', 'nodejs-centos7-node12' ]
+                image: [ 'nodejs-centos6-default', 'nodejs-centos7-node12', 'nodejs-centos7-node14' ]
```
Comment (sfc-gh-pmotacki) [inline .github/workflows/build-test.yml:112]: Do we still need centos6 here?

Comment (sfc-gh-dprzybysz) [reply .github/workflows/build-test.yml:112]: I'm going to cleanup centos6 and node 12 builds in separate commits

**Review by sfc-gh-pmotacki — APPROVED** [review]
LGTM

**Pull request merged**